libraries/freertos_plus/standard/https/include/iot_https_client.h

```diff
@@ -344,6 +409,131 @@ IotHttpsReturnCode_t IotHttpsClient_SendSync(IotHttpsConnectionHandle_t connHand
+IotHttpsReturnCode_t IotHttpsClient_SendAsync(IotHttpsConnectionHandle_t connHandle, 
+                                              IotHttpsRequestHandle_t reqHandle, 
+                                              IotHttpsResponseHandle_t * pRespHandle, 
+                                              IotHttpsResponseInfo_t* pRespInfo);
+/* @[declare_https_client_sendasync] */
+
+/**
+ * @brief Cancel an Asynchronous request.
+ * 
+ * This will stop an asynchronous request.
+ * 
+ * If this is called before the scheduled asynchronous request actually runs, then request will not be sent. 
+ * If this is called during any of the asynchronous callbacks, then the request/response will stop processing when the 
+ * callback returns. This is useful for any error conditions, found during the asynchronous callbacks, where the
+ * application wants to stop the rest of the request processing.
+ * 
+ * If the asynchronous request stops processing, the buffers configured in #IotHttpsResponseInfo_t.userBuffer and 
+ * #IotHttpsRequestInfo_t.userBuffer can be freed, modified, or reused only after the 
+ * #IotHttpsClientCallbacks_t.responseCompleteCallback in invoked.
+ * 
+ * <b> Example Asynchronous Code </b>
+ * @code{c}
+ * void _applicationDefined_appendHeaderCallback(void * pPrivData, IotHttpsRequestHandle_t reqHandle)
+ * {
+ *      char token[MAX_TOKEN_LENGTH] = { 0 }
+ *      int len = MAX_TOKEN_LENGTH;
+ *      int status = gen_auth_token(token, &len);
+ *      if( status == GEN_TOKEN_FAIL)
+ *      {
+ *          IotHttpsClient_CancelRequestAsync(reqHandle);
+ *      }
+ *      ...
+ * }
+ * 
+ * void _applicationDefined_writeCallback(void * pPrivData, IotHttpsRequestHandle_t reqHandle)
+ * {
+ *      if( application_data_get(writeBuffer, writeBufferLen) == GEN_TOKEN_FAIL)
+ *      {
+ *          IotHttpsClient_CancelRequestAsync(reqHandle);
+ *      }
+ *      ...
+ * }
+ * @endcode
+ * 
+ * @param[in] reqHandle - Request handle associated with the request.
+ * 
+ * @return One of the following:
+ * - IOT_HTTPS_OK if the request was successfully cancelled.
+ */
+/* @[declare_https_client_cancelrequestasync] */
+IotHttpsReturnCode_t IotHttpsClient_CancelRequestAsync(IotHttpsRequestHandle_t reqHandle);
+/* @[declare_https_client_cancelrequestasync] */
+
+/**
+ * @brief Cancel an Asynchronous response.
+ * 
+ * This will stop an asynchronous response.
+ * 
+ * If this is called during ANY of the asynchronous callbacks, then the response will stop processing when the callback
+ * returns. This is useful for any error conditions, found during the asynchronous callbacks, where the application 
+ * wants to stop the rest of the response processing.
+ * 
+ * If the asynchronous request stops processing, the buffers configured in #IotHttpsResponseInfo_t.userBuffer and 
+ * #IotHttpsRequestInfo_t.userBuffer can be freed, modified, or reused only after the 
+ * #IotHttpsClientCallbacks_t.responseCompleteCallback in invoked.
```

Comment:
why do we have 2 functions which do similar things? When would the app developer call one vs other?

Reply:
When in the callback and only the reqHandle is available.
When in the callback and only the respHandle is available we call the other.